### PR TITLE
1.實作遊戲結束功能

### PR DESCRIPTION
**Pull request by ted791029**
## Why need this change? / Root cause: 
- end-game
## Changes made:
- Room related
## Test Scope / Change impact:
- RoomController
## Issue
- #184

### CODE REVIEW

File: domain/src/main/kotlin/tw/waterballsa/gaas/domain/Room.kt [hunk 1]
```diff
@@ -83,6 +94,14 @@ class Room(
 
     private fun findPlayer(playerId: Player.Id): Player? = players.find { it.id == playerId }
 
+    private fun playersCancelReady() {
```
Comment (frankvicky) [inline domain/src/main/kotlin/tw/waterballsa/gaas/domain/Room.kt:97]: `cancelReadyForNonHostPlayers`

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt [hunk 2]
```diff
@@ -493,6 +517,15 @@ class RoomControllerTest @Autowired constructor(
...
 
+    private fun givenPlayersArePlayingInRoom(host: Player, vararg players: Player): Room {
+        val combinedPlayers = (listOf(host) + players).toMutableList()
+        players.forEach { player ->
+            player.ready()
+        }
+        testRoom = createRoom(host = host, players = combinedPlayers, status = PLAYING)
+        return testRoom
+    }
+
```
Comment (frankvicky) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt:528]:
```kotlin
    private fun givenPlayersArePlayingInRoom(host: Player, vararg players: Player): Room {
        players.forEach { it.ready() }
        val allPlayers = mutableListOf(host, *players)
        testRoom = createRoom(host = host, players = allPlayers, status = PLAYING)
        return testRoom
    }
```

**Review by frankvicky — CHANGES_REQUESTED** [review]

File: domain/src/main/kotlin/tw/waterballsa/gaas/domain/Room.kt (hunk 1 above)
Comment (ted791029) [reply domain/src/main/kotlin/tw/waterballsa/gaas/domain/Room.kt:97]: 已修正

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt (hunk 2 above)
Comment (ted791029) [reply spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt:528]: 已修正

**Review by frankvicky — APPROVED** [review]

**Pull request merged**